### PR TITLE
Adds join tables and queries:

### PR DESCRIPTION
**Pull request by HabibPayenda**
### **The following requirements were added to the project:**

- Create a table named vets with the following columns:
- [ ] id: integer (set it as autoincremented PRIMARY KEY)
- [ ] name: string
- [ ] age: integer
- [ ] date_of_graduation: date

- There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.
- There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.

- Insert the following data for vets:
- [ ] Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
- [ ] Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
- [ ] Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
- [ ] Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

- Insert the following data for specialties:
- [ ] Vet William Tatcher is specialized in Pokemon.
- [ ] Vet Stephanie Mendez is specialized in Digimon and Pokemon.
- [ ] Vet Jack Harkness is specialized in Digimon.

- Insert the following data for visits:
- [ ] Agumon visited William Tatcher on May 24th, 2020.
- [ ] Agumon visited Stephanie Mendez on Jul 22th, 2020.
- [ ] Gabumon visited Jack Harkness on Feb 2nd, 2021.
- [ ] Pikachu visited Maisy Smith on Jan 5th, 2020.
- [ ] Pikachu visited Maisy Smith on Mar 8th, 2020.
- [ ] Pikachu visited Maisy Smith on May 14th, 2020.
- [ ] Devimon visited Stephanie Mendez on May 4th, 2021.
- [ ] Charmander visited Jack Harkness on Feb 24th, 2021.
- [ ] Plantmon visited Maisy Smith on Dec 21st, 2019.
- [ ] Plantmon visited William Tatcher on Aug 10th, 2020.
- [ ] Plantmon visited Maisy Smith on Apr 7th, 2021.
- [ ] Squirtle visited Stephanie Mendez on Sep 29th, 2019.
- [ ] Angemon visited Jack Harkness on Oct 3rd, 2020.
- [ ] Angemon visited Jack Harkness on Nov 4th, 2020.
- [ ] Boarmon visited Maisy Smith on Jan 24th, 2019.
- [ ] Boarmon visited Maisy Smith on May 15th, 2019.
- [ ] Boarmon visited Maisy Smith on Feb 27th, 2020.
- [ ] Boarmon visited Maisy Smith on Aug 3rd, 2020.
- [ ] Blossom visited Stephanie Mendez on May 24th, 2020.
- [ ] Blossom visited William Tatcher on Jan 11th, 2021.

- Write queries to answer the following:
- [ ] Who was the last animal seen by William Tatcher?
- [ ] How many different animals did Stephanie Mendez see?
- [ ] List all vets and their specialties, including vets with no specialties.
- [ ] List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- [ ] What animal has the most visits to vets?
- [ ] Who was Maisy Smith's first visit?
- [ ] Details for most recent visit: animal information, vet information, and date of visit.
- [ ] How many visits were with a vet that did not specialize in that animal's species?
- [ ] What specialty should Maisy Smith consider getting? Look for the species she gets the most.

- Take a screenshot of the results of your queries.
- Remember to add the screenshot with the results of your queries to your Pull Request description.

### **Screenshoots:**

Please check if I have missed something.

**Thank you.**

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -37,5 +37,33 @@ UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle', 'Blosso
 UPDATE animals SET owner_id = 5 WHERE name IN ('Angemon', 'Boarmon');
 COMMIT;
 
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('William Tatcher', 45, '2000-7-23');
```
Comment (mahbubul14) [inline data.sql]: - [ ] Kindly fix the graduation month for William Tatcher.
  `Project requirements: Vet William Tatcher is 45 years old and graduated on Apr 23rd, 2000`

File: data.sql [hunk 2]
```diff
@@ -37,5 +37,33 @@ UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle', 'Blosso
...
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Maisy Smith', 26, '2019-1-17');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Stephanie Mendez', 64, '1981-3-4');
```
Comment (mahbubul14) [inline data.sql]: - [ ] Kindly fix the graduation month for Stephanie Mendez.
  `Project requirements: Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.`

File: data.sql [hunk 3]
```diff
@@ -37,5 +37,33 @@ UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle', 'Blosso
...
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Jack Harkness', 38, '2008-5-8');
```
Comment (mahbubul14) [inline data.sql]: - [ ] Kindly fix the graduation month for Jack Harkness.
  `Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.`

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @HabibPayenda  ,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: data.sql (hunk 3 above)
Comment (HabibPayenda) [reply data.sql]: @mahbubul14 Thank you

File: data.sql [hunk 4]
```diff
@@ -37,5 +37,33 @@ UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle', 'Blosso
 UPDATE animals SET owner_id = 5 WHERE name IN ('Angemon', 'Boarmon');
 COMMIT;
 
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('William Tatcher', 45, '2000-4-23');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Maisy Smith', 26, '2019-1-17');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Stephanie Mendez', 64, '1981-4-4');
```
Comment (tresorsawasawa) [inline data.sql]: - [ ] As mentioned by the previous reviewer, kindly make sure you fix the `date_of_graduation` for Vet **'Stephanie Mendez'**. The correct date is `1981-05-04` and not `1981-04-04`

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @HabibPayenda 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @HabibPayenda 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**